File: examples/python/remote_gateway_chat.py
from __future__ import annotations

from cogentlm import ChatMessage

from _common import (
    CogentClient,
    add_gateway_remote,
    print_text,
    read_remote_args,
    text_options,
)


def main() -> None:
    alias, prompt = read_remote_args("Explain remote inference in one sentence.")
    client = CogentClient()
    endpoint = add_gateway_remote(client, alias)
    run = client.chat(
        [
            ChatMessage("system", "Answer concisely."),
            ChatMessage("user", prompt),
        ],
        endpoint=endpoint,
        options=text_options(),
        emit_tokens=True,
    )
    streamed = ""
    for batch in run.tokens():
        print(batch["text"], end="", flush=True)
        streamed += batch["text"]
    print()
    result = run.result()
    if streamed != result["text"]:
        raise RuntimeError("streamed token batches did not match final response text")
    print_text(result)


if __name__ == "__main__":
    main()

File: examples/rust/src/bin/remote_gateway_chat.rs
use cogentlm::engine::{ChatMessage, ChatRole};
use cogentlm::{CogentChatRequest, CogentClient};
use cogentlm_rust_examples::remote_common;
use futures::executor::block_on;
use futures::StreamExt;

fn main() -> remote_common::ExampleResult<()> {
    block_on(async {
        let args = remote_common::args("Explain remote inference in one sentence.")?;
        let mut client = CogentClient::new();
        let endpoint = client.add_remote(
            args.alias.clone(),
            remote_common::gateway_remote(args.alias)?,
        )?;
        let run = client.chat(CogentChatRequest {
            endpoint: Some(endpoint),
            messages: vec![
                ChatMessage::new(ChatRole::System, "Answer concisely."),
                ChatMessage::new(ChatRole::User, args.input),
            ],
            options: remote_common::text_options(),
            emit_tokens: true,
            ..Default::default()
        });
        let (mut tokens, response) = run.into_parts();
        let mut streamed = String::new();
        while let Some(batch) = tokens.next().await {
            print!("{}", batch.text);
            streamed.push_str(&batch.text);
        }
        println!();
        let response = response.await?;
        assert_eq!(streamed, response.text);
        remote_common::print_text(response);
        Ok(())
    })
}
